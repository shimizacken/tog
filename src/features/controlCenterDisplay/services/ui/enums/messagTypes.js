const MessageTypes = {
    DRIVE: 'DRIVE',
    SYSTEM: 'SYSTEM',
    ALERT: 'ALERT'
};

const AlertSeverityTypes = {
    INFO: 'INFO',
    LOW: 'LOW',
    WARNING: 'WARNING',
    HIGH: 'HIGH',
    CRITICAL: 'CRITICAL' 
};

export {
    MessageTypes,
    AlertSeverityTypes
};
