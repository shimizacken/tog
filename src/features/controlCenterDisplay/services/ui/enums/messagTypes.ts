export enum MessageTypes {
  DRIVE = "DRIVE",
  SYSTEM = "SYSTEM",
  ALERT = "ALERT",
}

export enum AlertSeverityTypes {
  INFO = "INFO",
  LOW = "LOW",
  WARNING = "WARNING",
  HIGH = "HIGH",
  CRITICAL = "CRITICAL",
}
