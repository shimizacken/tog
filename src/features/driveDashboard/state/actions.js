import { SET_DRIVE_BUTTON_STATUS, SET_THROTTLE_SPEED, SET_DRIVE_BUTTON_SWITCHING_STATUS } from "./constants";
import DrivingButtonStates from "../services/ui/enums/drivingButtonStates";

export const setDrivingButtonState = drivingButtonState => ({
  type: SET_DRIVE_BUTTON_STATUS,
  drivingButtonState
});

export const setDrivingButtonSwitchingState = () => ({
  type: SET_DRIVE_BUTTON_SWITCHING_STATUS,
  drivingButtonState: DrivingButtonStates.SWITCHING
});

export const setThrottleSpeed = throttleSpeed => ({
  type: SET_THROTTLE_SPEED,
  throttleSpeed,
  meta: {
    throttle: 500
  }
});
