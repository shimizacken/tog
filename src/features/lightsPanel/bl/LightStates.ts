export enum LightStates {
  OFF = "OFF",
  ON = "ON",
  ERROR = "ERROR",
}
