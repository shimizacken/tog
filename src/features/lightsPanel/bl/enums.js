export const LightStates = {
    OFF: 'OFF',
    ON: 'ON',
    ERROR: 'ERROR'
};
