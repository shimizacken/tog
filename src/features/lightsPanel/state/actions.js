import { SET_FRONT_LIGHT_STATE, SET_BACK_LIGHT_STATE, SET_CABIN_LIGHT_STATE, SET_WAGON_LIGHT_STATE } from "./constants";

export const setFrontLightState = frontLightsState => ({
  type: SET_FRONT_LIGHT_STATE,
  frontLightsState
});

export const setBackLightState = backLightsState => ({
    type: SET_BACK_LIGHT_STATE,
    backLightsState
});

export const setCabinLightState = cabinLights => ({
    type: SET_CABIN_LIGHT_STATE,
    cabinLights
});

export const setWagonsLightState = wagonsLights => ({
    type: SET_WAGON_LIGHT_STATE,
    wagonsLights
});
