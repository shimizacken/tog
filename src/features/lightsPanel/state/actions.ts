import {
  SET_FRONT_LIGHT_STATE,
  SET_BACK_LIGHT_STATE,
  SET_CABIN_LIGHT_STATE,
  SET_WAGON_LIGHT_STATE,
} from "./constants";

export const setFrontLightState = {
  type: SET_FRONT_LIGHT_STATE,
};

export const setBackLightState = {
  type: SET_BACK_LIGHT_STATE,
};

export const setCabinLightState = {
  type: SET_CABIN_LIGHT_STATE,
};

export const setWagonsLightState = {
  type: SET_WAGON_LIGHT_STATE,
};
