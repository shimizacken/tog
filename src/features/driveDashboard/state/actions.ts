import {
  SET_DRIVE_BUTTON_STATUS,
  SET_THROTTLE_SPEED,
  SET_DRIVE_BUTTON_SWITCHING_STATUS,
  SET_DRIVING_STATUS,
} from "./constants";
import { DrivingButtonStates } from "../services/ui/enums/drivingButtonStates";

export interface SetDrivingButtonState {
  type: string;
  drivingButtonState: DrivingButtonStates;
}

export interface SetDrivingButtonSwitchingState {
  type: string;
  throttleSpeed: number;
  meta: {
    throttle: number;
  };
}

export interface SetDrivingStatus {
  type: string;
  drivingStatus: boolean;
}

export const setDrivingButtonState = (
  drivingButtonState: DrivingButtonStates
): SetDrivingButtonState => ({
  type: SET_DRIVE_BUTTON_STATUS,
  drivingButtonState,
});

export const setDrivingButtonSwitchingState = (): SetDrivingButtonState => ({
  type: SET_DRIVE_BUTTON_SWITCHING_STATUS,
  drivingButtonState: DrivingButtonStates.SWITCHING,
});

export const setThrottleSpeed = (
  throttleSpeed: number
): SetDrivingButtonSwitchingState => ({
  type: SET_THROTTLE_SPEED,
  throttleSpeed,
  meta: {
    throttle: 10,
  },
});

export const setDrivingStatus = (drivingStatus: boolean): SetDrivingStatus => ({
  type: SET_DRIVING_STATUS,
  drivingStatus,
});
