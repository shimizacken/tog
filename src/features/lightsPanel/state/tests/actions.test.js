import { setFrontLightState } from "../actions";
import { LightStates } from "../../bl/enums";

describe('lights actions tests', () => {
  
    it('should turn setFrontLightState on', () => {
        
        const frontLightsState = LightStates.ON

        expect(setFrontLightState(frontLightsState)).toMatchSnapshot();
    });

    it('should turn setFrontLightState off', () => {
        
        const frontLightsState = LightStates.OFF

        expect(setFrontLightState(frontLightsState)).toMatchSnapshot();
    });
});
