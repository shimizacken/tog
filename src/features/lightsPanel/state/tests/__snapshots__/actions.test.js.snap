// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lights actions tests should return setBackLightState 1`] = `
Object {
  "type": "SET_BACK_LIGHT_STATE",
}
`;

exports[`lights actions tests should return setCabinLightState action 1`] = `
Object {
  "type": "SET_CABIN_LIGHT_STATE",
}
`;

exports[`lights actions tests should return setFrontLightState action 1`] = `
Object {
  "type": "SET_FRONT_LIGHT_STATE",
}
`;

exports[`lights actions tests should return setWagonsLightState action 1`] = `
Object {
  "type": "SET_WAGON_LIGHT_STATE",
}
`;
